lib/email/receiver.rb

```diff
@@ -765,23 +765,44 @@ def create_group_post(group, user, body, elided)
           .order(created_at: :desc)
           .limit(1)
           .pluck(:post_id).last
-        post_ids << post_id_from_email_log
+        post_ids << post_id_from_email_log if post_id_from_email_log
       end
 
       if post_ids.any? && post = Post.where(id: post_ids).order(:created_at).last
 
-        # this must be done for the unknown user (who is staged) to
-        # be allowed to post a reply in the topic
-        if group.allow_unknown_sender_topic_replies
-          post.topic.topic_allowed_users.find_or_create_by!(user_id: user.id)
-        end
+        # We don't want users reopening months-old PM discussions in
+        # group SMTP inbox environments. Instead we just create a new
+        # topic with a link back to the old one.
+        if destination_too_old?(post) && group.smtp_enabled
+          enable_email_pm_setting(user)
```

Comment:
this line is non obvious, what is going on here?

Reply:
See https://github.com/discourse/discourse/blob/dfbac403fa367fc801bd86280f202d9fbf0302ec/lib/email/receiver.rb#L1324-L1329, we also do it when the user is normally creating a topic via email to the group https://github.com/discourse/discourse/blob/dfbac403fa367fc801bd86280f202d9fbf0302ec/lib/email/receiver.rb#L786-L786 

I might just move it into `create_topic` because we have a similar call in `create_reply`:

https://github.com/discourse/discourse/blob/56a7ad0a8f2e9e5be3117e2cd2a537319b2cec11/lib/email/receiver.rb#L1090-L1090

Reply:
yeah moving it to create_topic may work especially if create_reply already has it.